### PR TITLE
[CapMan visibility] fix throttling mechanism

### PR DESCRIPTION
**Pull request by xurui-c**
The throttling mechanism was incorrect. This PR makes it correct

### CODE REVIEW

File: tests/query/allocation_policies/test_bytes_scanned_rejecting_policy.py [hunk 1]
```diff
@@ -107,53 +107,6 @@ def test_consume_quota(
     assert allowance.can_run
 
 
-@pytest.mark.redis_db
```
Comment (volokluev) [inline tests/query/allocation_policies/test_bytes_scanned_rejecting_policy.py:110]: why are you removing this test?

Comment (xurui-c) [reply tests/query/allocation_policies/test_bytes_scanned_rejecting_policy.py:110]: it is covered by `test_does_not_throttle_and_then_throttles` and `test_does_not_throttle_and_then_throttles` has more realistic numbers

File: snuba/query/allocation_policies/bytes_scanned_rejecting_policy.py [hunk 2]
```diff
@@ -257,7 +257,17 @@ def _get_quota_allowance(
...
+            customer_tenant_key,
+            customer_tenant_value,
+        ) = self._get_customer_tenant_key_and_value(tenant_ids)
+        if (
+            granted_quota.granted
+            < self.__get_scan_limit(
+                customer_tenant_key, customer_tenant_value, referrer
+            )
+            - throttle_threshold
+        ):
```
Comment (volokluev) [inline snuba/query/allocation_policies/bytes_scanned_rejecting_policy.py:270]: Add a comment to explain what you're doing here so someone can easily understand it

File: tests/query/allocation_policies/test_bytes_scanned_rejecting_policy.py [hunk 3]
```diff
@@ -291,3 +244,63 @@ def test_penalize_timeout(policy: BytesScannedRejectingPolicy) -> None:
...
+        pytest.param(
+            {
+                "project_id": 4505240668733440,
+                "referrer": "api.trace-explorer.stats",
+            },
+            100000001,
+            200000000,
+        ),
+    ],
+)
```
Comment (volokluev) [inline tests/query/allocation_policies/test_bytes_scanned_rejecting_policy.py:261]: You probably don't need to parametrize this, there's only one example and it's testing a specific behavior

**Review by volokluev — APPROVED** [review]
Feel free to merge when tests pass

**Pull request merged**